src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixPlugin.groovy

```diff
@@ -50,16 +50,15 @@ class ScalafixPlugin implements Plugin<Project> {
 
         project.tasks.named('check').configure { it.dependsOn checkTask }
 
+        if (!extension.autoConfigureSemanticdb) { // TODO remove in a future version
+            logger.warn("WARNING: the 'autoConfigureSemanticdb' property is deprecated. Please use 'semanticdb.autoconfigure' instead")
+        }
+
         project.sourceSets.each { SourceSet sourceSet ->
             if (ScalaSourceSet.isScalaSourceSet(project, sourceSet) && !extension.ignoreSourceSets.get().contains(sourceSet.name)) {
                 def scalaSourceSet = new ScalaSourceSet(project, sourceSet)
-
-                if (scalaSourceSet.scalaVersion.present) {
-                    configureTaskForSourceSet(project, scalaSourceSet, IN_PLACE, fixTask, extension)
-                    configureTaskForSourceSet(project, scalaSourceSet, CHECK, checkTask, extension)
-                } else {
-                    logger.warn("WARNING: Skipping source set '${sourceSet.name}' as the Scala version could not be detected")
-                }
```

Comment:
the condition in this IF statement was always triggering dependency resolution during the configuration phase